File: C#-OOP-Basics/Problems Course Instance 2018/DefiningClasses/CompanyRoster/Employee.cs
public class Employee
{
    public Employee()
    {
        this.Email = "n/a";
        this.Age = -1;
    }

    public Employee(string name, decimal salary, string position, string department) : this()
    {
        this.Name = name;
        this.Salary = salary;
        this.Position = position;
        this.Department = department;
    }

    public string Name { get; set; }

    public decimal Salary { get; set; }

    public string Position { get; set; }

    public string Department { get; set; }

    public string Email { get; set; }

    public int Age { get; set; }
}

File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson01DefiningClasses/Exercise/pr04_company_roster/model/Employee.java
package Lesson01DefiningClasses.Exercise.pr04_company_roster.model;

public class Employee implements Comparable<Employee> {
    private Person person;
    private double salary;
    private String position;
    private String department;
    private String email;

    public Employee(Person person, double salary, String position, String department, String email) {
        super();
        this.person = person;
        this.salary = salary;
        this.position = position;
        this.department = department;
        this.email = email;
    }

    public String getEmployeeInfo() {
        return String.format("%s %.2f %s %d",
                this.person.getName(),
                this.salary,
                this.email,
                this.person.getAge());
    }

    @Override
    public int compareTo(Employee o) {
        return Double.compare(o.salary, this.salary);
    }

    public double getSalary() {
        return salary;
    }
}
